packages/iobio-react-components/src/utils/scoreFileTools.mts

```diff
@@ -0,0 +1,106 @@
+/*
+ *
+ * Copyright (c) 2025 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ * This program and the accompanying materials are made available under the terms of
+ * the GNU Affero General Public License v3.0. You should have received a copy of the
+ * GNU Affero General Public License along with this program.
+ * If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+import urlJoin from 'url-join';
+import {
+	fileMetaDataSchema,
+	type FileDocument,
+	type FileMetaData,
+	type ScoreDownloadParams,
+} from './scoreFileTypes.ts';
+
+const baseScoreDownloadParams: Omit<ScoreDownloadParams, 'length'> = {
+	external: 'true',
+	offset: '0',
+	'User-Agent': 'unknown',
+};
+
+// Iobio File Properties
+export const bamFileExtension = 'BAM';
+export const cramFileExtension = 'CRAM';
+export const BamFileExtensions = [bamFileExtension, cramFileExtension];
+
+/** Request File from Score API */
+export const getScoreFile = async ({
+	length,
+	objectId,
+	scoreApiUrl,
+	scoreApiDownloadPath,
+}: {
+	length: string;
+	objectId: string;
+	scoreApiUrl: string;
+	scoreApiDownloadPath: string;
+}): Promise<FileMetaData | undefined> => {
+	if (!(scoreApiUrl && scoreApiDownloadPath)) {
+		throw new Error('Score API URL is missing in .env');
+	}
+	const scoreDownloadParams: ScoreDownloadParams = {
+		...baseScoreDownloadParams,
+		length,
+	};
+	const urlParams = new URLSearchParams(scoreDownloadParams).toString();
+	try {
+		const scoreUrl = urlJoin(scoreApiUrl, scoreApiDownloadPath, objectId, `?${urlParams}`);
+		const response = (await fetch(scoreUrl)).json();
```

Comment:
Combining .json() onto this line doesn't actually address the issue @ciaranschutte was pointing out, this now returns the Promise from the .json deserialization to the caller and requires them to handle those errors. If you want any errors from the .json call to be handled by the catch block in this function then you need to await the .json() call as well:

```ts
const response = await fetch(scoreUrl);
const jsonData = await response.json();
return jsonData;
```